src/codeu/chat/client/commandline/Chat.java

```diff
@@ -45,11 +48,15 @@ public Chat(Context context) {
   // is willing to take another command, the function will return true. If
   // the system wants to exit, the function will return false.
   //
-  public boolean handleCommand(String line) {
+  public boolean handleCommand(String line) throws IOException {
 
-    final Scanner tokens = new Scanner(line.trim());
-
-    final String command = tokens.hasNext() ? tokens.next() : "";
+    final List<String> args = new ArrayList<String>();
+    Tokenizer tokenizer = new Tokenizer(line);
+    for (String token = tokenizer.next(); token != null; token = tokenizer.next()) {
+      args.add(token);
+    }
+    final String command = args.get(0);
+    args.remove(0);
```

Comment:
[`List.remove`](http://docs.oracle.com/javase/8/docs/api/java/util/List.html#remove-int-) also returns the element at that position before it was removed, so you can condense these two lines into one:
```java
final String command = args.remove(0);
```

(edit: looks like you already know this, as shown by later code, so I'm left scratching my head as to why this is here.)